systemtest/src/test/java/io/strimzi/systemtest/specific/RackAwarenessST.java

```diff
@@ -183,6 +284,70 @@ void testMirrorMaker2RackAwareness(ExtensionContext extensionContext) {
         String hostname = podNodeName.contains(".") ? podNodeName.substring(0, podNodeName.indexOf(".")) : podNodeName;
         String commandOut = cmdKubeClient(testStorage.getNamespaceName()).execInPod(podName, "/bin/bash", "-c", "cat /tmp/strimzi-connect.properties | grep consumer.client.rack").out().trim();
         assertThat(commandOut.equals("consumer.client.rack=" + hostname), is(true));
+
+        // Mirroring messages by: Producing to the Source Kafka Cluster and consuming them from mirrored KafkaTopic in target Kafka Cluster.
+
+        resourceManager.createResource(extensionContext, KafkaTopicTemplates.topic(kafkaClusterSourceName, testStorage.getTopicName(), 3).build());
+
+        LOGGER.info("Producing messages into the source Kafka: {}/{}, topic: {}", testStorage.getNamespaceName(), kafkaClusterSourceName, testStorage.getTopicName());
+        KafkaClients clients = new KafkaClientsBuilder()
+            .withProducerName(testStorage.getProducerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(kafkaClusterSourceName))
+            .withNamespaceName(testStorage.getNamespaceName())
+            .withTopicName(testStorage.getTopicName())
+            .withMessageCount(testStorage.getMessageCount())
+            .build();
+        resourceManager.createResource(extensionContext, clients.producerStrimzi());
+        ClientUtils.waitForProducerClientSuccess(testStorage);
+
+        LOGGER.info("Consuming messages in the target Kafka: {}/{} mirrored topic: {}", testStorage.getNamespaceName(), kafkaClusterTargetName, sourceMirroredTopicName);
+        clients = new KafkaClientsBuilder(clients)
+            .withTopicName(sourceMirroredTopicName)
+            .withConsumerName(testStorage.getConsumerName())
+            .withBootstrapAddress(KafkaResources.plainBootstrapAddress(kafkaClusterTargetName))
+            .build();
+        resourceManager.createResource(extensionContext, clients.consumerStrimzi());
+        ClientUtils.waitForConsumerClientSuccess(testStorage);
+    }
+
+    private void produceConsumeDataWithSinkConnector(ExtensionContext extensionContext, TestStorage testStorage) {
```

Comment:
The name is a bit misleading - you are:

1. deploying KafkaConnector
2. checking that KafkaConnector is ready
3. sending + receiving messages
4. checking that messages are in file sink

I would maybe split the method or change the name to more clear one